src/sql/src/statements.rs

```diff
@@ -238,11 +238,14 @@ fn parse_column_default_constraint(
 // a column is not nullable but has a default value null.
 /// Create a `ColumnSchema` from `ColumnDef`.
 pub fn column_def_to_schema(column_def: &ColumnDef, is_time_index: bool) -> Result<ColumnSchema> {
-    let is_nullable = column_def
-        .options
-        .iter()
-        .any(|o| matches!(o.option, ColumnOption::Null));
-
+    let is_nullable = if column_def.options.is_empty() {
```

Comment:
It might be better to add more tests for this function. Currently, we only cover this function in [alter's tests](https://github.com/GreptimeTeam/greptimedb/blob/ca5734edb34c201d5cbdff3bb29a15bd897d1d47/src/datanode/src/sql/alter.rs#L112) and [create's tests](https://github.com/GreptimeTeam/greptimedb/blob/ca5734edb34c201d5cbdff3bb29a15bd897d1d47/src/datanode/src/sql/create.rs#L254). It would be more convenient to test it in this file.